core/src/signal.rs

```diff
@@ -1,14 +1,47 @@
 use crate::action::ActionWrapper;
-use holochain_core_types::json::JsonString;
+use holochain_core_types::{error::HolochainError, json::JsonString};
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
 use std::{
     sync::mpsc::{channel, sync_channel, Receiver, SyncSender},
     thread,
 };
 
-#[derive(Debug)]
+#[derive(Clone, Debug, DefaultJson)]
 pub enum Signal {
     Internal(ActionWrapper),
     User(JsonString),
+    Holo(JsonString),
+}
+
+impl Serialize for Signal {
```

Comment:
Do we have to implement this by hand? If so, why do we?

Reply:
@sphinxc0re because we can't immediately derive `Serialize` for `Action`, which `Signal::Internal` contains. This is a temporary way to move forward without implementing Serialize for the very big `Action` enum.